service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/BaseDevice.java

```diff
@@ -188,34 +188,9 @@ public String getSecondaryKey()
      * @param primaryThumbprint the primary thumbprint to set
      * @param secondaryThumbprint the secondary thumbprint to set
      *
-     * @deprecated as of service-client version 1.15.1, please use {@link #setThumbprintFinal(String, String)}
-     *
-     * @throws IllegalArgumentException if primaryThumbprint or secondaryThumbprint is null or empty
-     */
-    @Deprecated
-    public void setThumbprint(String primaryThumbprint, String secondaryThumbprint)
-    {
-        if (Tools.isNullOrEmpty(primaryThumbprint) || Tools.isNullOrEmpty(secondaryThumbprint))
-        {
-            throw new IllegalArgumentException("Thumbprint may not be null or empty");
-        }
-
-        if (this.authentication == null)
-        {
-            this.authentication = new AuthenticationMechanism(AuthenticationType.SELF_SIGNED);
-        }
-
-        this.authentication.setPrimaryThumbprint(primaryThumbprint);
-        this.authentication.setSecondaryThumbprint(secondaryThumbprint);
-    }
-
-    /**
-     * Setter for X509 thumbprint
-     * @param primaryThumbprint the primary thumbprint to set
-     * @param secondaryThumbprint the secondary thumbprint to set
      * @throws IllegalArgumentException if primaryThumbprint or secondaryThumbprint is null or empty
      */
-    public final void setThumbprintFinal(String primaryThumbprint, String secondaryThumbprint)
+    public final void setThumbprint(String primaryThumbprint, String secondaryThumbprint)
```

Comment:
We suggested above that only setThumbprint is deprecated, not setThumbprintFinal. Would it not be a breaking change if we rename it?

Reply:
Yes, this is a breaking change. But this seems like a reasonable enough change since setters don't typically follow the pattern of ```setFooFinal()```